tests/numerics/ad/test_equation_system.py

```diff
@@ -82,3 +96,1266 @@ def test_evaluate_variables():
+
+    # Pass a Variable, we should get it back
+    var = setup.sd_top_variable.sub_vars[0]
+    received_variables_3 = eq_system._parse_variable_type([var])
+    assert len(received_variables_3) == 1
+    assert received_variables_3[0] == var
+
+    # Pass an md-variable with one sub-variable, check we get back the sub-variable
+    received_variables_4 = eq_system._parse_variable_type([setup.sd_top_variable])
+    assert len(received_variables_4) == 1
+    assert received_variables_4[0] == setup.sd_top_variable.sub_vars[0]
+
+    # Send an md-variable with two sub-variables
+    received_variables_5 = eq_system._parse_variable_type([setup.sd_variable])
+    assert len(received_variables_5) == len(setup.sd_variable.sub_vars)
+    assert all([var in received_variables_5 for var in setup.sd_variable.sub_vars])
+
+    # Send in the md-variable as a string, it should not make a difference
+    received_variables_6 = eq_system._parse_variable_type([setup.name_sd_variable])
+    assert len(received_variables_6) == len(setup.sd_variable.sub_vars)
+    assert all([var in received_variables_6 for var in setup.sd_variable.sub_vars])
+
+    # Send in two md-variables
+    received_variables_7 = eq_system._parse_variable_type(
+        [setup.sd_variable, setup.sd_top_variable]
+    )
+    assert len(received_variables_7) == len(setup.sd_variable.sub_vars) + len(
+        setup.sd_top_variable.sub_vars
+    )
+
+    # Send in a combination of string and variables.
+    # NOTE: While this combination goes against the specification in the type hints
+    # (list of strings and list of variable is permitted, combined lists are not), the
+    # current implementation actually allows for this. If the implementation is changed,
+    # the checks involving received_variables_8 and _9 can be deleted.
+    received_variables_8 = eq_system._parse_variable_type(
+        setup.sd_variable.sub_vars + [setup.name_sd_top_variable]
+    )
+    assert len(received_variables_8) == len(setup.sd_variable.sub_vars) + len(
+        setup.sd_top_variable.sub_vars
+    )
+
+    # Send in a combination of string and md-variables
+    received_variables_9 = eq_system._parse_variable_type(
+        [setup.sd_variable, setup.name_sd_top_variable]
+    )
+    assert len(received_variables_9) == len(setup.sd_variable.sub_vars) + len(
+        setup.sd_top_variable.sub_vars
+    )
+
+
+def test_parse_single_equation():
+    """Test the helper function for parsing single equations.
+
+    We consider only the equation posed on all subdomains, parsing of other equations
+    should be identical.
+
+    The test considers restrictions of the equation to subsets of its domains,
+    and verifies that the returned index sets are correctly ordered.
+
+    """
+    setup = EquationSystemSetup()
+    eq_system = setup.sys_man
+
+    # Represent the equation both by its string and its operator form.
+    # This could have been parametrized to the price of higher cost.
```

Comment:
```suggestion
    # This could have been parametrized to the price of computational higher cost 
    # (Pytest assembly overhead).
```

Reply:
Alternatively, use the setup fixture (default scope is function, which should be what we wish here), which should relieve most of the overhead. I think that may be preferable, as it will make debugging easier if this test fails: Then we'll know immediately if it fails for both operator and string version.